tempesta_fw/ss_skb.c

```diff
@@ -236,28 +236,25 @@ __it_next_data(struct sk_buff *skb, int i, TfwStr *it)
 static inline void
 __skb_insert_after(struct sk_buff *skb, struct sk_buff *nskb)
 {
-	SsSkbCb *scb = TFW_SKB_CB(skb);
-	SsSkbCb *nscb = TFW_SKB_CB(nskb);
-
-	nscb->next = scb->next;
-	nscb->prev = skb;
-	scb->next = nskb;
-	if (nscb->next)
-		TFW_SKB_CB(nscb->next)->prev = nskb;
+	nskb->next = skb->next;
+	nskb->prev = skb;
+	skb->next = nskb;
+	if (nskb->next)
+		nskb->next->prev = nskb;
 }
 
 /*
  * Update the skb list's pointer to the last item
  * if a new skb has been added at the end of the list.
  */
 static inline void
-__skb_skblist_fixup(SsSkbList *skb_list)
+__skb_skblist_fixup(struct sk_buff *skb_head)
 {
-	SsSkbCb *lscb = TFW_SKB_CB(skb_list->last);
+	struct sk_buff *last = skb_head->prev;
 
-	if (lscb->next)
-		skb_list->last = lscb->next;
-	WARN_ON_ONCE(TFW_SKB_CB(skb_list->last)->next);
+	if (last->next)
+		skb_head->prev = last->next;
+	WARN_ON_ONCE(skb_head->prev->next);
```

Comment:
I still don't see the reason for the function and why can't we move to double linked list if we need `last` pointer or accurate NULL terminated list like `skb->frag_list`: current `prev` meaning the last item as well as double linked list with NULLs for the last and first items look ugly and error prone.